### DISCUSSION TITLE
How to configure Master-Slave database using Doctrine in Symfony?

### DISCUSSION THREAD

**Discussion by fd6130** [symfony/symfony#44033] [Q&A]
Hi, is there a way to configure Master-Slave database using Doctrine in Symfony? I can't find any latest docs or tutorial about Master-Slave in Doctrine or Symfony (there is maybe, but outdated). I need that because I want to keep both database data in sync for blue-green deployment.

**Comment by fodger** [comment]
It should be good to have a real answer from Sensio team to know how to use simply replicas to send read requests, it's a common situation for pros.

**Comment by stof** [comment]
This is configured using the `replicas` setting of the DBAL configuration of DoctrineBundle. This is not called master-slave because DBAL moved away from that naming in favor of `PrimaryReadReplicaConnection` a few years ago.

**Comment by fodger** [comment]
Hi, 
I know but is there somewhere a real complete functional example to send select request on replica ?

I didn't find it anywhere in SF doc or Doctrine...

**Comment by fodger** [comment]
Can we have please a real functional example with replicas to send automatically SELECT requests on slave ?
It well helps a lot community.

Thanks a lot.

**Comment by fodger** [comment]
While Sensiolabs avoid this need ?

**Reply by stof** [reply]
Please read the replies. I already provided the answer in https://github.com/symfony/symfony/discussions/44033#discussioncomment-6699894

**Comment by fodger** [comment]
I saw that, but there is no real example about PrimaryReadReplicaConnection and replicas.
Must we use keep_replica option for example ? There's nothing official about it in SF doc.
It's boring to have rich functionalities without detailed doc.
